### ISSUE TITLE
Feature Request (dimensions)

### ISSUE THREAD

**Issue by Tolomaj** [nkallen/plasticity#26]
it would be super when there was method to set up dimensions with some type of dimensions control (like solidworks) or line properties.

**Comment by CGrrrr** [comment]
Agree.

**Comment by nkallen** [comment]
Maybe upload a screenshot of something that you have in mind? I'm thinking about dimensions, but I want to do something very lightweight. What works for concept artists, not traditional CAD.

**Comment by CGrrrr** [comment]
It would be nice if the dimensions were only available at the beginning of the object creation.

**Comment by Tolomaj** [comment]
> Maybe upload a screenshot of something that you have in mind? I'm thinking about dimensions, but I want to do something very lightweight. What works for concept artists, not traditional CAD.

i think about somethink like this.
![image](https://user-images.githubusercontent.com/59420562/156888102-4775ad3e-e1d0-40c0-a7b5-969d4d70161e.png)
when the user right-clicks on the line.
(its the only modified screansoth)

**Comment by CGrrrr** [comment]
Or there is a ruler function, which seems to be easier to implement. Like blender.
![ruler](https://user-images.githubusercontent.com/61892991/156904316-c8222982-36e9-4144-9279-918191ac53c7.png)

**Comment by nkallen** [comment]
I think I understand what you're asking for but .... But dimensions are not in plasticity because plasticity is a tool for concept artists. Sometimes more analytic concept artists want to work with precise measurements (a gun is exactly 10x3x2 units or whatever). But they don't need to dimension things..

**Comment by angellinares** [comment]
Hi, just popping in into the discussion.

As some of you say, in a concept art-oriented tool I don't see the need for dimensions (as the ones present in CAD packages), but more than dimensions I would consider the option of "published parameters" from objects. Those published parameters might later allow for linked parameters (two-dimensional parameters from 2 different geometries exposed and linked, so if one changes the other one does too). 

These published parameters might allow model tweaking from a dedicated palette and will speed up model tweaking or repurposing. Also, been persistent visual elements in the interface, might allow for a better understanding of proportions, etc. 

Applications that I might think of for these: a wheel model that could be tweaked for different vehicles, the barrel of a firearm that could be morphed to adapt to different models, etc.  

Cheers!

**Comment by ThomasPasieka** [comment]
I think it would be a big mistake to not include dimensions or the ability to precisely change dimensions and other values. Having measurements allows me to be more precise when positioning, scaling and moving pieces into place. It's not about precision but about the ability to make good design choices. Good designs are all about numbers and their relations within any given design/model. Giving us "dimensions" doesn't hurt your software. It empowers those who like to have more control over their designs so please reconsider your stance.

**Comment by Critical11** [comment]
Hmm maybe a toggle on or off if you don't want dimensions

**Comment by felle9900** [comment]
Tinkercad way is pretty pog. have to drop a ruler on ground to get dimensions

**Comment by Tolomaj** [comment]
> Hmm maybe a toggle on or off if you don't want dimensions
I don't think this feature needs a toggle, just a small [ADD DIMENSION] button that adds a dimension or angle rule to the edge, and you won't notice it until you use it.

**Comment by mainr** [comment]
I'd like to add my voice to the calls for basic dimensions in plasticity. I get @nkallen 's target audience is more 'concept only', but adding the _option_ to control dimensions doesn't detract at all from that goal. If you don't _need_ to control dimensions, then don't. I think that having a more abstract control over dimensions ( @angellinares ) like linking dimensions would also be beneficial (even to designers who are only looking for 'conceptual design'), however the technical details of implementing this control is likely to exceed that of basic dimension control.

I did just come across an interesting [video](https://www.youtube.com/watch?v=92QmjS-xDaI) ) showing a Blender add-on that is representative of an implementation, and supports the direction of add-ons to implement missing 'core' features - AS LONG AS the underlying code supports the requirement & exposes the required interfaces.
